verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -49,53 +50,169 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
+  verilog::FlowTree control_flow_tree(lexed_sequence);
+  auto status =
+      control_flow_tree
+          .GenerateControlFlowTree();  // construct the control flow tree.
+  status = control_flow_tree.DepthFirstSearch(
+      0);  // traverse the tree by dfs from the root (node 0).
+
+  // Printing the token streams of every possible variant.
+  int variants_counter = 1;
+  for (const auto& current_variant : control_flow_tree.variants_) {
+    outs << "Variant number " << variants_counter++ << ":\n";
+    for (auto token : current_variant) outs << token << '\n';
+    puts("");
+  }
+
+  return absl::OkStatus();
+}
+
+ABSL_FLAG(bool, strip_comments, false,
+          "Replaces comments with white-spaces in files passed.");
+ABSL_FLAG(bool, multiple_cu, true,
+          "Files are preprocessed in separate compilation units.");
+ABSL_FLAG(bool, generate_variants, false,
+          "Generates every possible variants w.r.t. compiler conditionals");
+
+int main(int argc, char* argv[]) {
+  const std::string usage =
+      absl::StrCat("usage:\n", argv[0], " [options] file [<file>...]\n\n",
+                   "options summary:\n",
+                   "-multiple_cu: Files are preprocessed in separate "
+                   "compilation units.\n",
+                   "-strip_comments: Replaces one/multi-line comments with "
+                   "equal white-spaces.\n",
+                   "-generate_variants: Generates every possible variants "
+                   "w.r.t. compiler conditionals.\n");
 
   // Process invocation args.
   const auto args = verible::InitCommandLine(usage, &argc, &argv);
-  if (args.size() == 1) {
+  if (args.empty()) {
     std::cerr << absl::ProgramUsageMessage() << std::endl;
     return 1;
   }
-  // args[0] is the program name
-  // args[1] is the subcommand
-  // subcommand args start at [2]
-  const SubcommandArgsRange command_args(args.cbegin() + 2, args.cend());
-
-  const auto& sub = commands.GetSubcommandEntry(args[1]);
-  // Run the subcommand.
-  const auto status = sub.main(command_args, std::cin, std::cout, std::cerr);
-  if (!status.ok()) {
-    std::cerr << status.message() << std::endl;
+
+  // Get flags.
+  bool strip_comments_flag = absl::GetFlag(FLAGS_strip_comments);
+  bool multiple_cu_flag = absl::GetFlag(FLAGS_multiple_cu);
+  bool generate_variants_flag = absl::GetFlag(FLAGS_generate_variants);
+
+  // Check for flags and argument illegal usage.
+  if (strip_comments_flag && generate_variants_flag) {  // Illegal usage.
+    std::cerr << "ERROR: the flags passed shouldn't be used together.\n\n"
+              << absl::ProgramUsageMessage() << std::endl;
     return 1;
   }
+  if (args.size() ==
+      1) {  // No Files are passed (files are passed as positional arguments).
+    std::cerr << "ERROR: No System-Verilog files were passed.\n\n"
+              << absl::ProgramUsageMessage() << std::endl;
+    return 1;
+  }
+
+  // Select the operation mode and execute it.
+  if (strip_comments_flag) {
```

Comment:
this nested if/else if with for loop and another if inside is tedious to read. Suggest having separate small functions that are called on the various flags.